### PR TITLE
Bump build number (and update CI)

### PR DESCRIPTION
**Pull request by jakirkham**
Tries to fix CI to adjust for the fact that NumPy ABI pinning happens now and prep for a new release.

### CODE REVIEW

**Comment by jakirkham** [comment]
Lifted the CI changes from `rank_filter` as I was having trouble generating them. Probably just don't know the right commands to do. In any event, this seem to be working ok from what I can tell. So, am merging this ASAP as it is causing trouble for others trying to use `h5py`.

File: .travis.yml [hunk 1]
```diff
@@ -7,9 +7,12 @@ language: objective-c
 env:
   matrix:
     
-    - CONDA_PY=27  
-    - CONDA_PY=34  
-    - CONDA_PY=35  
+    - CONDA_NPY=110  CONDA_PY=27
```
Comment (pelson) [inline .travis.yml:10]: Ideally, try to avoid editing this file by hand. Just run `conda smithy rerender` on the feedstock...

Comment (jakirkham) [inline .travis.yml:10]: Yeah, I tried, but it was complaining about missing CI tokens. As it was quite late and I wanted to get something out to fix a problem that someone was having when using `h5py`, I picked the quickly and dirty route of making the change manually. Maybe we can chat about the issue I was running into and see if there was a usage error on my part or something that needs fixing.

Comment (pelson) [inline .travis.yml:10]: > Maybe we can chat about the issue I was running into and see if there was a usage error on my part or something that needs fixing.

Ah, yes. Thanks @jakirkham. I'd say it was something that could be improved with conda-smithy. You shouldn't need tokens to re-render.

Comment (pelson) [inline .travis.yml:10]: @jakirkham - I just tried and it seems that conda-smithy doesn't need the tokens (though it does tell you you don't have them). That is late ones for you :sleeping: :smile:

Comment (ocefpaf) [inline .travis.yml:10]: Older versions did raise errors due to missing tokens. @jakirkham what version of conda-smithy do you have installed? (Although due to the recent improvements you should do directly to the latest version.)

Comment (pelson) [inline .travis.yml:10]: Agreed. It would be unhealthy to rerender using an older conda-smithy IMO.

Comment (jakirkham) [inline .travis.yml:10]: I think I had 0.6.0 at the time. Looks like there have been a few releases since.

**Pull request merged**